### PR TITLE
Feat/add settings modal to folder card

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR adds a settings dropdown and folder renaming modal to the existing FolderCard. This allows users to rename collections/resource categories, or delete collections/resource categories with all their associated contents.

In addition, this PR is also intended to address the following issues:
- fix misc text overflow issues, and remove resource room name from the settings, to resolve https://github.com/isomerpages/isomercms-frontend/issues/236
- fix the page moving modal not closing when clicked outside, to resolve https://github.com/isomerpages/isomercms-frontend/issues/244

### CODE REVIEW

File: src/components/FolderModal.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            title={displayText}
+            id="newCategoryName"
+            value={newCategoryName}
+            onFieldChange={folderNameChangeHandler}
+          />
+          <div className={elementStyles.modalButtons}>
+            <SaveDeleteButtons
+              isDisabled={false}
+              hasDeleteButton={false}
+              saveCallback={saveHandler}
```
Comment (kwajiehao) [inline src/components/FolderModal.jsx:51]: Can we align these to the right?

File: src/components/FolderModal.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      </div>
+    </div>
+  )
+};
+
+FolderModal.propTypes = {
+  displayTitle: PropTypes.string.isRequired,
+  displayText: PropTypes.string.isRequired,
+  onProceed: PropTypes.func.isRequired,
+  onCancel: PropTypes.func.isRequired,
```
Comment (kwajiehao) [inline src/components/FolderModal.jsx:64]: These props are marked as `isRequired` but aren't defined on the component

File: src/components/FolderCard.jsx [hunk 3]
```diff
@@ -13,38 +21,108 @@ const FolderCard = ({
...
+          siteName={siteName}
+          isCollection={isCollection}
+        />
+      }
+      { canShowDeleteWarningModal &&
+        <DeleteWarningModal
+          onCancel={() => setCanShowDeleteWarningModal(false)}
+          onDelete={deleteHandler}
+          type="folder"
+        />
```
Comment (kwajiehao) [inline src/components/FolderCard.jsx:84]: This should have been changed before, but can we also align the delete warning modal's buttons to the right?

**Review by kwajiehao — CHANGES_REQUESTED** [review]
left very minor comments

File: src/components/FolderModal.jsx (hunk 2 above)
Comment (alexanderleegs) [reply src/components/FolderModal.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/245/commits/697b4c7cbed36050b2a91809324cf3fc3fb5200b

File: src/components/FolderModal.jsx (hunk 1 above)
Comment (alexanderleegs) [reply src/components/FolderModal.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/245/commits/583bedc98151581efcf6afe1a21147cf9d1bbcc9

File: src/components/FolderCard.jsx (hunk 3 above)
Comment (alexanderleegs) [reply src/components/FolderCard.jsx:84]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/245/commits/583bedc98151581efcf6afe1a21147cf9d1bbcc9

**Review by kwajiehao — APPROVED** [review]
lgtm

**Comment by kwajiehao** [comment]
Note that this PR also fixes the first point in issue @232

**Pull request merged**